ladybug/datacollection.py

```diff
@@ -163,16 +162,27 @@ def get_highest_values(data, count):
             highest_values_index: Indicies of the n highest values in data 
                 list, ordered from highest to lowest
         """
-        lenght_data = len(data)
+        dataPoints = self._data
+        
+        values = list(obj._value for obj in dataPoints)
+        
+        lenght_values = len(values)
         
         count = int(count)
         
-        highest_values = sorted(data, reverse = True)[0:count]
+        assert count <= lenght_values, \
+            'Count must be equal to or smaller than list of data lenght'
+            
+        assert count > 0, \
+            'Count must be higher than zero'
+        
+        self.highest_values = sorted(values, reverse = True)[0:count]
```

Comment:
I don't think you need to assign `highest_values` to class as a property. This is a once tot be used data and also can change based on input for `count`.